x/foundation/keeper/msg_server.go

```diff
@@ -297,3 +297,43 @@ func (s msgServer) Revoke(c context.Context, req *foundation.MsgRevoke) (*founda
 
 	return &foundation.MsgRevokeResponse{}, nil
 }
+
+// OneTimeMint defines a method to withdraw coins from the treasury.
+func (s msgServer) OneTimeMint(c context.Context, req *foundation.MsgOneTimeMint) (*foundation.MsgOneTimeMintResponse, error) {
+	ctx := sdk.UnwrapSDKContext(c)
+
+	oneTimeMintCount := s.keeper.GetOneTimeMintCount(ctx)
+	if oneTimeMintCount == 0 {
+		return nil, sdkerrors.ErrUnauthorized.Wrapf("The one-time-mint can no longer be executed.")
+	}
+
+	if err := s.keeper.validateOperator(ctx, req.Operator); err != nil {
+		return nil, err
+	}
+
+	if req.Amount.Empty() {
+		return nil, sdkerrors.ErrUnauthorized.Wrapf("The one-time-mint request amount is empty.")
+	}
+
```

Comment:
It should be moved to `ValidateBasic()` of the corresponding msg.

Reply:
I moved that codes. thx.